.github/workflows/test_docker_build.yml

```diff
@@ -4,11 +4,35 @@ jobs:
   test:
     runs-on: ubuntu-latest
     steps:
-      - name: 'Checkout git repo'
-        uses: actions/checkout@v1
-      - name: Build Docker image
-        uses: docker/build-push-action@v1.0.1
+      - name: Checkout git repo
```

Comment:
can we maybe get rid of this file since it is almost identical to .github/workflows/publish_docker_image.yml ?

Reply:
And then maybe set the `push` flag conditionally based on the event?